File: exp-10/q1-bfs/solution.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	adj := make([][]int, n+1)
	for i := 0; i < m; i++ {
		var u, v int
		fmt.Scan(&u, &v)
		adj[u] = append(adj[u], v)
		adj[v] = append(adj[v], u)
	}
	var s int
	fmt.Scan(&s)

	for i := 1; i <= n; i++ {
		sort.Ints(adj[i])
	}

	visited := make([]bool, n+1)
	queue := []int{s}
	visited[s] = true
	order := []int{}
	for len(queue) > 0 {
		u := queue[0]
		queue = queue[1:]
		order = append(order, u)
		for _, v := range adj[u] {
			if !visited[v] {
				visited[v] = true
				queue = append(queue, v)
			}
		}
	}
	fmt.Printf("BFS Traversal: ")
	for i, u := range order {
		if i > 0 {
			fmt.Printf(" ")
		}
		fmt.Printf("%d", u)
	}
	fmt.Println()
}

File: exp-10/q1-bfs/Solution.java
import java.util.*;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        List<List<Integer>> adj = new ArrayList<>();
        for (int i = 0; i <= n; i++)
            adj.add(new ArrayList<>());
        for (int i = 0; i < m; i++) {
            int u = sc.nextInt();
            int v = sc.nextInt();
            adj.get(u).add(v);
            adj.get(v).add(u);
        }
        int s = sc.nextInt();

        for (int i = 1; i <= n; i++)
            Collections.sort(adj.get(i));

        boolean[] visited = new boolean[n + 1];
        Queue<Integer> q = new LinkedList<>();
        visited[s] = true;
        q.add(s);

        System.out.print("BFS Traversal: ");
        boolean first = true;
        while (!q.isEmpty()) {
            int u = q.poll();
            if (!first)
                System.out.print(" ");
            first = false;
            System.out.print(u);
            for (int v : adj.get(u)) {
                if (!visited[v]) {
                    visited[v] = true;
                    q.add(v);
                }
            }
        }
        System.out.println();
        sc.close();
    }
}